packages/@aws-cdk/service-spec-importers/src/importers/import-cloudformation-registry.ts

```diff
@@ -123,17 +127,45 @@ export function importCloudFormationRegistryResource(options: LoadCloudFormation
           return schemaTypeToModelType(nameHint, jsonschema.setResolvedReference(combinedType, reference), fail);
         }
 
-        const convertedTypes = inner.map((t) => schemaTypeToModelType(nameHint, resolve(t), fail));
+        let convertedTypes;
+        if (jsonschema.isOneOf(resolvedSchema)) {
+          // There are two ways `oneOf` can be used:
+          // 1. (Problematic Case) The property item itself is one of multiple object types,
```

Comment:
I think you explained it to me, but I do not remember .. could you share some example